site/en/docs/extensions/mv3/nativeMessaging/index.md

```diff
@@ -0,0 +1,202 @@
+a separate process and communicates with it using standard input and standard output streams.
+
+## Native messaging host {: #native-messaging-host }
+
+In order to register a native messaging host the application must install a manifest file that
+defines the native messaging host configuration. Below is an example of the manifest file:
+
+```json
+{
+  "name": "com.my_company.my_application",
+  "description": "My Application",
+  "path": "C:\\Program Files\\My Application\\chrome_native_messaging_host.exe",
+  "type": "stdio",
+  "allowed_origins": ["chrome-extension://knldjmfmopnpolahpmmgbagdohdnhkik/"]
+}
+```
+
+The native messaging host manifest file must be valid JSON and contains the following fields:
+
+<table class="simple" id="native-messaging-host-manifest"><tbody><tr><th>Name</th><th>Description</th></tr><tr><td><code>name</code></td><td>Name of the native messaging host. Clients pass this string to <a href="/apps/runtime#method-connectNative">runtime.connectNative</a> or <a href="/apps/runtime#method-sendNativeMessage">runtime.sendNativeMessage</a>. This name can only contain lowercase alphanumeric characters, underscores and dots. The name cannot start or end with a dot, and a dot cannot be followed by another dot.</td></tr><tr><td><code>description</code></td><td>Short application description.</td></tr><tr><td><code>path</code></td><td>Path to the native messaging host binary. On Linux and macOS the path must be absolute. On Windows it can be relative to the directory in which the manifest file is located. The host process is started with the current directory set to the directory that contains the host binary. For example if this parameter is set to <code>C:\Application\nm_host.exe</code> then it will be started with current directory <code>C:\Application\</code>.</td></tr><tr><td><code>type</code></td><td>Type of the interface used to communicate with the native messaging host. Currently there is only one possible value for this parameter: <code>stdio</code>. It indicates that Chrome should use <code>stdin</code> and <code>stdout</code> to communicate with the host.</td></tr><tr><td><code>allowed_origins</code></td><td>List of extensions that should have access to the native messaging host. Wildcards such as <code>chrome-extension://*/*</code> are <em>not</em> allowed.</td></tr></tbody></table>
+
+### Native messaging host location {: #native-messaging-host-location }
+
+The location of the manifest file depends on the platform.
+
+On **Windows**, the manifest file can be located anywhere in the file system. The application
+installer must create registry key
+`HKEY_LOCAL_MACHINE\SOFTWARE\Google\Chrome\NativeMessagingHosts\com.my_company.my_application` or
+`HKEY_CURRENT_USER\SOFTWARE\Google\Chrome\NativeMessagingHosts\com.my_company.my_application`, and
+set default value of that key to the full path to the manifest file. For example, using the
+following command:
+
+```text
```

Comment:
I'm reluctant to set bash here since this is a windows server command, not bash, and the highlighting might not be correct. Text is nice because it just renders plainly.